pygeos/constructive.py

```diff
@@ -255,6 +256,46 @@ def centroid(geometry, **kwargs):
     return lib.centroid(geometry, **kwargs)
 
 
+@multithreading_enabled
+def clip_by_rect(geometry, xmin, ymin, xmax, ymax, **kwargs):
+    """
+    Returns the portion of a geometry within a rectangle.
+
+    The geometry is clipped in a fast but possibly dirty way. The output is
+    not guaranteed to be valid. No exceptions will be raised for topological
+    errors.
+
```

Comment:
Based on the results for empty and your test cases here, it might be a good idea to note that empty geometries or geometries that do not overlap with the specified bounds will result in `GEOMETRYCOLLECTION EMPTY`.